include/lsst/afw/typehandling/detail/refwrap_utils.h

```diff
@@ -0,0 +1,127 @@
+// -*- LSST-C++ -*-
+/*
+ * This file is part of afw.
+ *
+ * Developed for the LSST Data Management System.
+ * This product includes software developed by the LSST Project
+ * (https://www.lsst.org).
+ * See the COPYRIGHT file at the top-level directory of this distribution
+ * for details of code ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef LSST_AFW_TYPEHANDLING_DETAIL_REFWRAP_UTILS_H
+#define LSST_AFW_TYPEHANDLING_DETAIL_REFWRAP_UTILS_H
+
+#include <functional>
+#include <variant>
+
+namespace lsst {
+namespace afw {
+namespace typehandling {
+namespace detail {
+
+// An equality comparison function object for std::variant visitors that
+// contain std::reference_wrapper types.  Because reference_wrapper doesn't
+// define operator==, std::variant::operator== can't do this for us.
+struct refwrap_equals {
+
+    template <typename T>
+    bool operator()(std::reference_wrapper<T> a, std::reference_wrapper<T> b) {
+        return a.get() == b.get();
+    }
+
+    template <typename T, typename U>
```

Comment:
I realize this shouldn't be instantiated if `T == U`, but can you add an `enable_if` as an extra safeguard?